cmd/argocd-applicationset-controller/commands/applicationset_controller.go

```diff
@@ -241,6 +242,7 @@ func NewCommand() *cobra.Command {
 	command.Flags().StringVar(&scmRootCAPath, "scm-root-ca-path", env.StringFromEnv("ARGOCD_APPLICATIONSET_CONTROLLER_SCM_ROOT_CA_PATH", ""), "Provide Root CA Path for self-signed TLS Certificates")
 	command.Flags().StringSliceVar(&globalPreservedAnnotations, "preserved-annotations", env.StringsFromEnv("ARGOCD_APPLICATIONSET_CONTROLLER_GLOBAL_PRESERVED_ANNOTATIONS", []string{}, ","), "Sets global preserved field values for annotations")
 	command.Flags().StringSliceVar(&globalPreservedLabels, "preserved-labels", env.StringsFromEnv("ARGOCD_APPLICATIONSET_CONTROLLER_GLOBAL_PRESERVED_LABELS", []string{}, ","), "Sets global preserved field values for labels")
+	command.Flags().IntVar(&webhookParallelism, "webhook-parallelism-limit", env.ParseNumFromEnv("ARGOCD_APPLICATIONSET_CONTROLLER_WEBHOOK_PARALLELISM_LIMIT", 25, 1, 1000), "Number of webhook requests processed concurrently")
```

Comment:
1000 is set as max limit to avoid footguns. If a large setup has say 5K Applications, and each application is 4KB (from what I see), 1000 concurrent processing will use 5K * 4KB * 1K = 20GB of memory. I think that already is a very large number. If you still think its less, I can set it to more, maybe 5K or 10K?

Do you have any guidance on default values? I can set it to 50 or 100.

> Also we should document this so that others are aware of this setting.
I have noted this flag in `argo-server.md`. Could you point out where else would you like this.

Reply:
@gdsoumya I have updated the default to 50. I also see you are talking about AppSet flag documentation, but I don't see any obvious place to document this. Maybe in the appset webhook section in the git generator?

https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Generators-Git/#webhook-configuration